processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -888,25 +931,34 @@ private static int compareDimsInRowsWithAggs(
     }
   }
 
+  static long estimateStringKeySize(String key)
+  {
+    return (long) key.length() * Chars.BYTES + ROUGH_OVERHEAD_PER_DICTIONARY_ENTRY;
```

Comment:
+ char array header, + `String` object

Reply:
Yeah, I just copied as it was and that might be more precise estimation. It would be nice to do in https://github.com/druid-io/druid/issues/4768.